### PR TITLE
Partial typing of ``imports.py``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

### CODE REVIEW

File: pylint/checkers/imports.py [hunk 1]
```diff
@@ -536,7 +549,17 @@ def leave_module(self, node: nodes.Module) -> None:
...
+    def compute_first_non_import_node(
+        self,
+        node: nodes.If
+        | nodes.Expr
+        | nodes.Comprehension
+        | nodes.IfExp
+        | nodes.Assign
+        | nodes.AssignAttr
+        | nodes.TryExcept
+        | nodes.TryFinally,
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/imports.py:561]: If this can be everything except an import node it could also be a classDef or a lot of things, right ? Maybe ``NodeNG`` would make sense ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Maybe we could create an alias for ``nodes.Import | nodes.ImportFrom`` ?

File: pylint/checkers/imports.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/checkers/imports.py:561]: See L577. I thought I would be explicit just like we normally are for `visit_` methods.

That said L622 shows that L602 is incorrect.

I'm drafting as I'd like to fix that in this PR.

**Comment by DanielNoord** [comment]
> Maybe we could create an alias for `nodes.Import | nodes.ImportFrom` ?

There is `mixins.ImportFrom` in `astroid` but since that doesn't inherit from `NodeNG` it doesn't really work.

Let's put https://github.com/PyCQA/astroid/issues/1633 in `2.12` and do the necessary refactoring as well. Marc just gave his thumbs up for the issue so I think he is also okay with my proposal to start inheriting from `NodeNG` for the `mixins`.

TLDR: Drafting as we need `astroid` `2.12` with additional changes + a fix for `visit_functiondef` in this module.

**Review by Pierre-Sassoulas — APPROVED** [review]
If mypy is happy, I'm happy too 😄
